### PR TITLE
18 00 sw error handling

### PR DESCRIPTION
**Pull request by diana-macrometa**
This PR updates the stream worker error handling section based on [Jira ticket #603](https://macrometa.atlassian.net/browse/CEP-603) and [Jira ticket #601](https://macrometa.atlassian.net/browse/CEP-601).

### CODE REVIEW

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (StoyanTc) [inline docs/cep/error-handling/error-handling-wait-and-retry.md]: @diana-macrometa - Might be the annotation should be `OnError.action='wait'`?

Comment (grainier-macrometa) [reply docs/cep/error-handling/error-handling-wait-and-retry.md]: @diana-macrometa - @StoyanTc is correct. Since this section talks about `Wait and Retry Example` the action should be `wait`.

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (grainier-macrometa) [inline docs/cep/error-handling/error-handling-wait-and-retry.md]: @diana-macrometa - Here as well, it should be `wait`

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (grainier-macrometa) [inline docs/cep/error-handling/error-handling-wait-and-retry.md]:
```suggestion
```

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (grainier-macrometa) [inline docs/cep/error-handling/error-handling-wait-and-retry.md]: Here also `wait`

File: docs/cep/error-handling/index.md
Comment (grainier-macrometa) [inline docs/cep/error-handling/index.md]:
```suggestion
For sinks, the same `OnError.action='stream'` property can be used to capture and handle errors. This ensures that any issues during the event publication process are properly managed.
```

**Comment by grainier-macrometa** [comment]
@diana-macrometa - `stream` is just a one action we can use with error handling. We support following actions for following elements:

- Table
	- wait
	- stream

- Sink
	- log
	- wait
	- stream

- Other elements
	- log
	- stream

`wait` - Wait for sometime (i.e. 5 seconds) and retry.
`stream` - Forward the error and the event to fault stream.
`log` - Logs the error, and drops the message.

It's better If we can put this information somewhere as well.

cc: @StoyanTc @rahulprabhune-macrometa

**Comment by diana-macrometa** [comment]
Updated content to include the other actions. Please make sure the syntax shown works for all three. I suspect info is missing, especially for things like the amount of time to wait and retry.

**Comment by diana-macrometa** [comment]
@grainier-macrometa and @StoyanTc, please review

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (grainier-macrometa) [inline docs/cep/error-handling/error-handling-wait-and-retry.md]: @diana-macrometa / @StoyanTc - Do we have a `mock logger service` to download?

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (StoyanTc) [inline docs/cep/error-handling/error-handling-wait-and-retry.md]: @diana-macrometa - The customer cannot use this endpoint the service's for two reasons:

1. He has not an access to the `c8cep` localhost.
2. If he wants to run such service somewhere else he needs the previous explanations you have already moved out below.

You could restore the lines for the logger's service and replaces `http://localhost:8080/logger` by `http|https://<loggers' host>:8080|443/logger`. All this depends on whether we should mention about this `siddhi` service at all.

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (diana-macrometa) [reply docs/cep/error-handling/error-handling-wait-and-retry.md]: Not that I am aware of. Sorry, that should not have been removed. Fixed.

**Review by StoyanTc — COMMENTED** [review]
@diana-macrometa - My only comment is about the `logger` service. However I'm not certain about the recent implementation. Please don't merge the PR without @grainier-macrometa 's approval.

File: docs/cep/error-handling/table-error-handling.md
Comment (grainier-macrometa) [inline docs/cep/error-handling/table-error-handling.md]:
```suggestion
@App:name("test-db")
```

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (diana-macrometa) [reply docs/cep/error-handling/error-handling-wait-and-retry.md]: Whenever possible, we should avoid mentioning siddhi.

I do not understand what changes you are asking for.

**Review by grainier-macrometa — APPROVED** [review]
Added a comment and a suggestion. Other than that, LGTM.

**Comment by diana-macrometa** [comment]
@StoyanTc, is this good to go, or do we need more changes?

File: docs/cep/error-handling/error-handling-wait-and-retry.md
Comment (StoyanTc) [reply docs/cep/error-handling/error-handling-wait-and-retry.md]: @diana-macrometa - If we'd like to have full-fledged example you should restore the section with the logger service no matter that it includes `siddhi`.
The customer could run the service anywhere, even on his local machine/laptop and access it from GDN's stream worker.
However the url cannot be `http://localhost:8080/logger` but e.g. `http://<customer's host>:8080/logger` or ``https://<customer's host>:443/logger`.

Comment (diana-macrometa) [reply docs/cep/error-handling/error-handling-wait-and-retry.md]: Updated, and I put the prerequisite link back in.

**Review by StoyanTc — APPROVED** [review]

**Pull request merged**
